articles/Good-Engineering-Practices-While-Working-Solo.md

```diff
@@ -7,132 +7,124 @@
 
 ---
 
-# Good Engineering Practices while Working Solo
+# 当你独自研发时，你可以这么做
 
-When you’ve got to go it alone, how do you make the most out of it?
+当你独自研发（Working solo）时，你该如何充分发挥自己？
 
-Most developers work as part of a team. However, at some point in our careers, we’ve had to (or we’ll have to) work alone. And while much of product development involves being able to manage or work with the rest of the team, it’s equally as important to develop good practices while working solo.
+大部分开发者都在自己的团队里工作，不过有些时候，我们也会不得不独自研发，固然大量产品研发离不开团队的管理和协作，但是在个人研发时，遵循一些工程实践方法同样重要。
 
-## A quick note about “working solo”
+## 什么是 Working solo
 
-Solo typically means doing something alone. But we’re using it in this context to mean anything that’s done by a small number of people in an unofficial, unstructured environment. It could be just you, or you and a few other folks. Examples include:
+Solo 一般指独自做某件事，但是在本文里，它特指一个人或一个小组在非办公环境下进行项目研发，例如：
 
-* An open-source project, such as a package or library
-* A personal product, which could be commercial or free
-* A freelance gig for a client
+* 开源项目，例如一个 package 或 library
+* 个人项目，可以是收费或免费的
+* 自由职业者
 
-The common thread here is that there are no established rules like you’d typically have in a company.
+上述示例的共同点是：你在开发时，并不会受到像公司里那样的约束。
 
-## Why should I bother about my engineering practices
+## 为什么需要关注工程实践
 
-Here are a few reasons why they matter:
+原因之一是**你将会成为团队不可或缺的财富。**
 
-**You’ll be an asset to your team.**
+我们设想一下你加入一个团队时最初的场景：
 
-Let’s consider the possible scenarios that could ensue when you join a team:
+* 你所加入的团队遵循的实践规范和你之前的一致，那么你会很快融入这个团队。
+* 两者不一致情况下，如果你之前遵循了最佳实践，那么用不了多久你也会融入。
+* 最糟糕情况是，如果你所在团队没有遵循最佳实践，你可以推广你的经验来改善团队流程，要不然只能离开了。
 
-* You join a team that follows the practices you’re used to. Great! It means you’ll be ready to start contributing from day one.
-* You join a team that follows a different set of practices. If you’ve grasped the general concept of good engineering practices, it shouldn’t take you too long to get used to it, and you’ll become productive soon, too.
-* You join a team that doesn’t follow any good practices (oh no!). In this case, depending on the team, you might be able to bring your knowledge to bear and improve their processes. Otherwise…maybe just leave.
+不管怎么说，我们期望的还是双赢。
 
-Either way, it’s a win-win.
+另外一个原因是**你还会成为一名更优秀的开发者。**
 
-**You’ll be a better developer.**
+软件工程可不止编程，一个产品从概念到上线需要考虑很多事情，遵循最佳实践可以帮助你走上正轨，少一些挫折。如果你钟情编码，那么当你做一些新事情时，总会有一种诱惑让你跳过其他事情直接埋头编码。但是随着时间的推移，我发现，遵循一些最佳实践会让我们在享受独自工作带来的灵活性的时候，同时帮助我们规避掉一些障碍。
```

Comment:
【If you love coding (like me), there’s always the temptation when working on something new to jump right in and start coding. But I’ve seen over time that having some kind of structure in place, while still maintaining the flexibility that comes with working solo, helps me to avoid many bumps in the road.】没有翻译